File: TP1/PYTHON/TP3/src/etudiant.py
from src.personne import Personne

class Etudiant(Personne):
    def __init__(self, nom, age, numero_etudiant, moyenne, cours):
        super().__init__(nom, age)
        self.numero_etudiant = numero_etudiant
        self.__moyenne = moyenne
        self.cours = cours
    
    def get_moyenne(self):
        return self.__moyenne
    
    def set_moyenne(self, moyenne):
        if moyenne < 0 or moyenne > 20:
            raise ValueError("Moyenne invalide")
        self.__moyenne = moyenne
    
    def afficher_details(self):
        return f"Etudiant: {self.get_nom()}, Age: {self.get_age()}, Moyenne: {self.get_moyenne()}"

File: TP1/JAVA/TP3/src/Etudiant.java
package src;
import src.Personne;
import src.Cours;

import java.util.List;

public class Etudiant extends Personne {

    private String numeroEtudiant;
    private double moyenne;
    private List<Cours> cours;

    public Etudiant(String nom, int age, String numeroEtudiant, double moyenne, List<Cours> cours) {
        super(nom, age);
        this.numeroEtudiant = numeroEtudiant;
        this.moyenne = moyenne;
        this.cours = cours;
    }

    public String getNumeroEtudiant() {
        return numeroEtudiant;
    }

    public double getMoyenne() {
        return moyenne;
    }

    public List<Cours> getCours() {
        return cours;
    }

    @Override
    public void afficherDetails() {
        System.out.println("Etudiant: " + getNom()
                + ", Age: " + getAge()
                + ", Moyenne: " + moyenne);
    }
}